### DISCUSSION TITLE
Enable code generation from strings in my Next.js app (production mode)

### DISCUSSION THREAD

**Discussion by w3villa-Aman-Paul** [vercel/community#6632] [Help]
### Summary

I am doing a project where I want to generate dynamic UI from the backend. In the current setup I want to send the UI code from my Python backend and the function should run the code in my Next.js app.
Here is the data which I am sending from my python backend:

> {
>         "prompt": "If you want to show events, call \`show_stock_price\` ", 
>         "message": "show me the stock price of AAPL.", 
>         "functions": [{
>                   "name": "show_stock_price",
>                   "description": "Get the current stock price of a given stock or currency. Use this to show the price to the user.",
>                   "parameters": {
>                       "type": "object",
>                       "properties": {
>                           "symbol": {
>                               "type": "string",
>                               "description": "The name or symbol of the stock or currency. e.g., DOGE/AAPL/USD."
>                           },
>                           "price": {
>                               "type": "number",
>                               "description": "The price of the stock."
>                           },
>                           "delta": {
>                               "type": "number",
>                               "description": "The change in price of the stock."
>                           }
>                       },
>                       "required": [
>                           "symbol",
>                           "price",
>                           "delta"
>                       ]
>                   }
>             }],
>         "action": """reply.update(React.createElement(BotCard, null, React.createElement(StockSkeleton, null)));await sleep(1000);reply.done(React.createElement(BotCard, null, React.createElement(Stock, { name: symbol_value, price: price_value, delta: delta_value})));aiState.done([...aiState.get(), { role: 'function', name: 'show_stock_price', content: '[Price of symbol_value = price_value]' }]);"""
>      }

     
and in my Next.js app I want to run the action code as follows:
   

>   completion.onFunctionCall(
>     functionName,
>     async (...params) => {
>       try {
>         let dynamicFunction = (Function("React", "reply", "sleep", "aiState", ...args || "", ...Object.keys(params[0]).map(key => `${key}_value`), stringfunction))(React, reply, sleep, aiState, ...resolvedArgs, ...Object.values(params[0]))
>         dynamicFunction()
>       } catch (error) {
>         console.log('Error in executing action code:', error);
>       }
>     },
>   );

The main issue is it is running fine in my development mode but in production mode it is showing this error:

> EvalError: Code generation from strings disallowed for this context
> at Function (<anonymous>)
> at Object.show_stock_price (/home/w3villa/Downloads/vercel-ai-demo/.next/server/edge-chunks/390.js:12:431)
> at Object.experimental_onFunctionCall (/home/w3villa/Downloads/vercel-ai-demo/.next/server/edge-chunks/390.js:12:7480)
> at Object.flush (/home/w3villa/Downloads/vercel-ai-demo/.next/server/edge-chunks/608.js:38:230185)
> at ensureIsPromise (node:internal/webstreams/util:185:19)
> at transformStreamDefaultSinkCloseAlgorithm (node:internal/webstreams/transformstream:569:5)
> at Object.close (node:internal/webstreams/transformstream:366:14)
> at ensureIsPromise (node:internal/webstreams/util:185:19)
> at writableStreamDefaultControllerProcessClose (node:internal/webstreams/writablestream:1144:28)
> at writableStreamDefaultControllerAdvanceQueueIfNeeded (node:internal/webstreams/writablestream:1224:5)

Please help me find a way to run this in production mode.

### Steps to Reproduce

1. Try to create a dynamic function using eval or the new Function constructor.
2. Create a build 
3. Run the build

**Comment by nhatdang2003** [comment]
My project has an error same. My production is working well. But the next day, I visit my production and I receive response error 500. I check my log and see the error: Code generation from strings disallowed for this context, and I think middleware make this error. I deploy my production on IIS of window server use iisnode.

[a comment by an automated account is not shown]
